mbed_build/_internal/config/config_action.py

```diff
@@ -0,0 +1,79 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""This module contains config actions, which will mutate configuration.
+
+Config actions are built from entries found in JSON files used by MbedOS applications.
+Some of those entries aren't simple overrides, but involve more complex operations
+like addition or removal of items from lists.
+"""
+from abc import ABC, abstractmethod, abstractclassmethod
+from dataclasses import dataclass
+from typing import Any, Optional
+
+# Fix circular dependency problem
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from mbed_build._internal.config.config import Config
```

Comment:
How come this is a circular dependency? Looks like config_action isn't imported by config.

Reply:
```
tests/_internal/config/test_config_action.py:7: in <module>
    from mbed_build._internal.config.config import Config
mbed_build/_internal/config/config.py:9: in <module>
    from mbed_build._internal.config.config_layer import ConfigLayer
mbed_build/_internal/config/config_layer.py:9: in <module>
    from mbed_build._internal.config.config_action import ConfigAction
mbed_build/_internal/config/config_action.py:19: in <module>
    from mbed_build._internal.config.config import Config
E   ImportError: cannot import name 'Config' from partially initialized module 'mbed_build._internal.config.config'
```

Reply:
Oh I see.